File: kits/python/lux/weather.py
import sys
import numpy as np

def get_weather_config(current_weather, env_cfg):
    if type(current_weather) == int or isinstance(current_weather, (int, np.integer)):
        current_weather = env_cfg.WEATHER_ID_TO_NAME[current_weather]
    if current_weather == "MARS_QUAKE":
        return dict(power_gain_factor=1, power_loss_factor=1)
    elif current_weather == "COLD_SNAP":
        return dict(power_gain_factor=1, power_loss_factor=env_cfg.WEATHER["COLD_SNAP"]["POWER_CONSUMPTION"])
    elif current_weather == "DUST_STORM":
        return dict(power_gain_factor=env_cfg.WEATHER["DUST_STORM"]["POWER_GAIN"], power_loss_factor=1)
    elif current_weather == "SOLAR_FLARE":
        return dict(power_gain_factor=env_cfg.WEATHER["SOLAR_FLARE"]["POWER_GAIN"], power_loss_factor=1)
    return dict(power_gain_factor=1, power_loss_factor=1)

File: kits/js/lux/weather.js
function getWeatherConfig(currentWeather, envCfg) {
  const currentWeatherName = envCfg['WEATHER_ID_TO_NAME'][currentWeather];
  let powerGainFactor = 1;
  let powerlossFactor = 1;
  if(currentWeatherName === "COLD_SNAP") {
    powerlossFactor = envCfg["WEATHER"]["COLD_SNAP"]["POWER_CONSUMPTION"];
  } else if(currentWeatherName === "DUST_STORM") {
    powerGainFactor = envCfg["WEATHER"]["DUST_STORM"]["POWER_GAIN"];
  } else if(currentWeatherName === "SOLAR_FLARE") {
    powerGainFactor = envCfg["WEATHER"]["SOLAR_FLARE"]["POWER_GAIN"];
  }
  return {powerGainFactor, powerlossFactor};
}

module.exports = { getWeatherConfig }
